### PR TITLE
Table sort improvements

### LINKED ISSUE #438: Table Sorting Improvements
* Add isSorted prop passed to HeaderCell to allow for custom styling.
* Enhance sort states on each column (ascending, descending, noSort) to allow for a three-click sort pattern.
* Add initialSortState field on Column to allow a column to be initially sorted.

### PR DESCRIPTION
**Pull request by xavierdonnellon**
This PR fixes #438 and accomplishes the following:
* Adds a third sort state – allows for toggling of the sort through ascending, descending, and the new unsorted state. This lets users disable a sort after they enable it if they want.
* Adds the `SortDirection` and `SortState` data types which clean up the code. 
* Passes the `isSorted` prop to `HeaderCell` and `ResponsiveHeaderCell` to allow for custom styling on the header when the column is being sorted by

### CODE REVIEW

File: src/components/Table/Table.tsx [hunk 1]
```diff
@@ -179,15 +181,28 @@ export const Cell = styled(StyledBaseTD)`
...
-    width: 1em;
-    transition: transform 0.2s, opacity 0.5s;
-    opacity: ${direction === null ? 0 : 1};
-    transform: rotate(${direction ? 0 : 180}deg);
-  `}
+  ${({ direction }: { direction?: SortDirection | boolean | null }) => {
+    if (typeof direction === 'boolean') {
+      // eslint-disable-next-line no-console
+      console.warn(
+        'From FoundryUI Table: Passing `direction` prop to `SortIcon` as a boolean is deprecated. Please pass in an argument of type `SortDirection` instead.',
```
Comment (aVileBroker) [inline src/components/Table/Table.tsx:192]:
```suggestion
        'From FoundryUI Table: Passing a boolean to the `direction` prop on `SortIcon` is deprecated. Use the `Table.SortDirection` enum instead.',
```

File: src/components/Table/Table.tsx [hunk 2]
```diff
@@ -282,8 +297,27 @@ const Table = ({
   groupLabelRowRef,
   headerRef,
 }: TableProps): JSX.Element => {
-  const [sortedData, sortData] = useState(data);
-  const [sortMethod, setSortMethod] = useState(defaultSort);
+  // Convert deprecated defaultSort argument if provided as [key, bool]
```
Comment (aVileBroker) [inline src/components/Table/Table.tsx:300]:
```suggestion
  // Convert defaultSort argument if provided in the deprecated form [key, bool]
```

File: src/components/Table/Table.tsx [hunk 3]
```diff
@@ -282,8 +297,27 @@ const Table = ({
...
+  if (Array.isArray(defaultSort)) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      'From FoundryUI Table: Passing `defaultSort` prop as type [string, boolean] is deprecated. Instead, pass an argument of type `SortState`.',
```
Comment (aVileBroker) [inline src/components/Table/Table.tsx:304]:
```suggestion
      'From FoundryUI Table: Passing `defaultSort` prop as type [string, boolean] is deprecated. Instead, pass an argument of type `Table.SortState`.',
```

File: src/components/Table/Table.tsx [hunk 4]
```diff
@@ -328,20 +362,26 @@ const Table = ({
    *
    * `direction` is true for ascending, false for descending.
    */
```
Comment (aVileBroker) [inline src/components/Table/Table.tsx:364]: this comment needs to be updated

File: src/components/Table/Table.tsx [hunk 5]
```diff
@@ -352,50 +392,66 @@ const Table = ({
...
   useEffect(() => {
-    handleOnSort(sortMethod[0], sortMethod[1]);
+    handleOnSort(sortState);
   }, [data]); // eslint-disable-line react-hooks/exhaustive-deps
 
+  const getNewSortState = (key: string): SortState => ({
+    sortedColumn: key,
+    // If we're changing sort state of an already-sorted column, cycle through direction. Otherwise, start at ascending.
+    direction:
+      sortState.sortedColumn === key ? (sortState.direction + 1) % 3 : SortDirection.ascending,
```
Comment (aVileBroker) [inline src/components/Table/Table.tsx:452]: lol

File: src/components/Table/__tests__/__snapshots__/Table.test.tsx.snap [hunk 6]
```diff
@@ -167,6 +169,7 @@ exports[`Table Shows Table with default props 1`] = `
           Numbers
           <svg
             class="c4"
+            direction="1"
```
Comment (aVileBroker) [inline src/components/Table/__tests__/__snapshots__/Table.test.tsx.snap:172]: the browser will complain that `direction` isn't a valid attribute of `svg`, I think there's a way to add `data-` prefix or something similar to make it happy. This needs to be done in other components but might as well test it here

File: src/components/Table/types.ts [hunk 7]
```diff
@@ -37,12 +37,28 @@ export type columnTypes = Columns;
 /** The data for a single row. */
 export type RowEntry = Record<string, any>;
 
+export interface SortState {
+  /** The key of the header column that is currently being sorted by. */
+  sortedColumn?: string;
```
Comment (aVileBroker) [inline src/components/Table/types.ts:42]: if it's not too late, we could change this to `sortedColumnKey` and get rid of the comment

File: src/components/Table/types.ts [hunk 8]
```diff
@@ -37,12 +37,28 @@ export type columnTypes = Columns;
...
+  direction: SortDirection;
+}
+
+export enum SortDirection {
+  noSort = 0,
+  ascending = 1,
+  descending = 2,
+}
```
Comment (aVileBroker) [inline src/components/Table/types.ts:50]: I believe enums do this automatically
```suggestion
export enum SortDirection {
  noSort,
  ascending,
  descending,
}
```

File: src/components/Table/types.ts [hunk 9]
```diff
@@ -37,12 +37,28 @@ export type columnTypes = Columns;
...
+
 export type TableProps = {
   areGroupsCollapsible?: boolean;
   columnGap?: string;
   columns: Columns;
   data?: Array<RowEntry> | Array<Array<RowEntry>>;
-  defaultSort?: [string, boolean]; // key, direction
+  /**
+   * Controls which field is initially sorted.
+   * The `[string, boolean]` type for this prop is deprecated. Instead, use `SortState`.
```
Comment (aVileBroker) [inline src/components/Table/types.ts:59]:
```suggestion
   * The `[string, boolean]` type for this prop is deprecated. Instead, use `Table.SortState`.
```

**Review by aVileBroker — CHANGES_REQUESTED** [review]

File: src/components/Table/Table.tsx (hunk 5 above)
Comment (xavierdonnellon) [reply src/components/Table/Table.tsx:452]: LOL indeed. Too cracked to remove

**Review by aVileBroker — APPROVED** [review]

**Pull request merged**